### PR TITLE
[WIP] Use pass-by-value on supported installations

### PR DESCRIPTION
**Pull request by zachasme**
Follow-up to #22.

This PR sets `h3index` type to be pass-by-value on supported systems.

It does so by using `LIKE = int8` for `CREATE TYPE` which should set `PASSEDBYVALUE` on relevant systems. It also defines new macros for fetching arguments and returning values that use `USE_FLOAT8_BYVAL` to determine whether to use pointers or not.

We need to look at two things before merging:

- [x] Figure out how to handle the upgrade path. It does not appear possible to alter the base type directly. Worst case we have to make a hard cut here and ask users to `DROP` and `CREATE` the extension again.
- [x] We need to update CI such that tests are run for both pass-by-value and -reference. Right now we can manually test by setting `ARCH=i386` in the Dockerfile, but Github Actions does not support Docker build-args :cry: 

Ping @Komzpa: Does this align with your original idea? And do you have any ideas for updating the datatype without dropping it first?

### CODE REVIEW

**Comment by Komzpa** [comment]
Hey, this looks exactly like my dreams! :)

For upgrades, I've seen this thing in PostGIS sources and I expect it to be relevant:
https://github.com/postgis/postgis/blob/master/postgis/postgis_after_upgrade.sql#L233

File: src/extension.in.h [hunk 1]
```diff
@@ -30,10 +30,37 @@ typedef struct
...
-#define PG_RETURN_H3_INDEX_P(x) PG_RETURN_POINTER(x)
-#define PG_GETARG_H3_INDEX_P(x) (H3Index *) PG_GETARG_POINTER(x)
+/*
+ * DatumGetH3Index
+ *		Returns H3 index value of a datum.
+ *
+ * Note: this macro hides whether h3 index is pass by value or by reference.
+ */
+
+#ifdef USE_FLOAT8_BYVAL
```
Comment (Komzpa) [inline src/extension.in.h:40]: I haven't checked, but if there is something like USE_INT8_BYVAL it might be irrelevantly more beautiful here.

Comment (zachasme) [reply src/extension.in.h:40]: I thought the same, but as far as I can tell `USE_FLOAT8_BYVAL` is the only flag used (including for int64).

**Comment by zachasme** [comment]
Very relevant! That solves the update path. Now I just have to set up CI for 32 bit as well.

@Komzpa I've removed all of the instances `PG_FREE_IF_COPY`. Am I right to assume they are no longer needed (and will error on pass-by-value anyway)?

**Pull request merged**